crates/subspace-farmer/src/dsn.rs

```diff
@@ -0,0 +1,26 @@
+use std::ops::Range;
+
+use futures::Stream;
+use subspace_core_primitives::PieceIndexHash;
+
+use crate::PiecesToPlot;
+
+#[async_trait::async_trait]
+pub trait DSNSync {
+    type Stream: Stream<Item = PiecesToPlot>;
+
+    async fn get_pieces(&mut self, range: Range<PieceIndexHash>) -> Self::Stream;
+}
```

Comment:
This kind of makes sense, but I can also imagine that we'd need an extra argument to actually validate the pieces we're receiving, so we can drop connections to peers that send us bad responses.

With that in mind I'm not sure whether single method is appropriate or there should be separate methods for finding peers and then requesting data from them.

I think eventually this should probably live in the networking instead.